lib/ecto/migrator.ex

```diff
@@ -30,10 +30,24 @@ defmodule Ecto.Migrator do
   """
   @spec migrations_path(Ecto.Repo.t) :: String.t
   def migrations_path(repo) do
+    hd(List.first(migrations_paths(repo)))
+  end
+
+  @doc """
+  Gets the migrations paths from a repository.
+  """
+  @spec migrations_paths(Ecto.Repo.t) :: [String.t]
+  def migrations_paths(repo) do
+    config = repo.config()
+    config[:migrations_paths] || [default_migration_path(repo)]
```

Comment:
Let's pass the config as argument to `default_migration_path` to avoid reading it twice.

Reply:
That's what I was originally doing but

https://github.com/elixir-ecto/ecto/blob/8cb17abc4d9914297b7ca8e3b5760c0ffc4b9fc7/lib/ecto/migrator.ex#L48

requires the repo to build the priv path.